kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -306,11 +293,16 @@ public static ConfigurationFragment<VirtualClusterGateway> buildVirtualClusterGa
         var volumes = tlsConfigFragment.volumes();
         var mounts = tlsConfigFragment.mounts();
 
+        NodeIdentificationStrategy nodeIdentificationStrategy = instance.nodeIdentificationStrategy();
+        PortIdentifiesNodeIdentificationStrategy portIdentifiesNodeIdentificationStrategy = nodeIdentificationStrategy instanceof PortIdentifiesNodeIdentificationStrategy
+                ? (PortIdentifiesNodeIdentificationStrategy) nodeIdentificationStrategy
+                : null;
+        SniHostIdentifiesNodeIdentificationStrategy sniHostIdentifiesNode = nodeIdentificationStrategy instanceof SniHostIdentifiesNodeIdentificationStrategy
+                ? (SniHostIdentifiesNodeIdentificationStrategy) nodeIdentificationStrategy
+                : null;
```

Comment:
blech, maybe ingress should have two methods like `Optional<Sni..> getSniStrategy()`, `Optional<PortIdent...> getPortyStrategy()`.

Reply:
I kept the logic out here but made it marginally less abhorrent